### PR TITLE
Add documentation for version contexts

### PR DESCRIPTION
**Pull request by attfarhan**
Add docs for version contexts and link from the dropdown + extension api docs.

### CODE REVIEW

File: doc/user/search/index.md [hunk 1]
```diff
@@ -92,6 +92,36 @@ Examples:
...
 
+
+### Version contexts
+
+Many companies have old versions of code running in production and need to search across all the code for a specific release.
+Version contexts allow creating sets of many repositories at specific revisions. When set, a version context limits your searches and code navigation actions (with basic code intelligence) to the repositories and revisions in the context.
+
+Your site admin can add version contexts in site configuration under the `experimentalFeatures.versionContexts` setting. For example:
+
+```
```
Comment (felixfbecker) [inline doc/user/search/index.md:103]:
```suggestion
```json
```

File: doc/user/search/index.md [hunk 2]
```diff
@@ -92,6 +92,36 @@ Examples:
...
+"experimentalFeatures": {
+  "versionContexts": [
+    {
+      "name": “Sourcegraph 3.16”,
+      "revisions": [
+        {
+          "repo": "github.com/sourcegraph/sourcegraph",
+          "ref": “v3.16.0”
```
Comment (felixfbecker) [inline doc/user/search/index.md:111]: This is probably too late - but we usualy call this `rev` (for revision, which is a git term), not `ref` (= reference?)

We also usually avoid abbreviations like `repo` (or `org`) in public APIs.

File: web/src/nav/VersionContextDropdown.tsx [hunk 3]
```diff
@@ -111,10 +111,10 @@ export const VersionContextDropdown: React.FunctionComponent<VersionContextDropd
                                         <div className="version-context-dropdown__info card">
                                             <span className="font-weight-bold">About version contexts</span>
                                             <p className="mb-2">
-                                                Version contexts allow you to search a set of repositories based on a
-                                                hash, tag, or other interesting moment in time of multiple code bases.
-                                                Your administrator can configure version contexts in the site
-                                                configuration.
+                                                Version contexts (<a href="http://docs.sourcegraph.com/user/search#version-contexts">documentation</a>) allow you to search a
+                                                set of repositories based on a hash, tag, or other interesting moment in
```
Comment (felixfbecker) [inline web/src/nav/VersionContextDropdown.tsx]:
```suggestion
                                                set of repositories based on a commit hash, tag, or other interesting moment in
```

Comment (felixfbecker) [reply web/src/nav/VersionContextDropdown.tsx]: How would I define an "other interesting moment in time"? Can I pass e.g. an ISO date?

File: doc/user/search/index.md [hunk 4]
```diff
@@ -92,6 +92,36 @@ Examples:
...
+### Version contexts
+
+Many companies have old versions of code running in production and need to search across all the code for a specific release.
+Version contexts allow creating sets of many repositories at specific revisions. When set, a version context limits your searches and code navigation actions (with basic code intelligence) to the repositories and revisions in the context.
+
+Your site admin can add version contexts in site configuration under the `experimentalFeatures.versionContexts` setting. For example:
+
+```
+"experimentalFeatures": {
+  "versionContexts": [
```
Comment (felixfbecker) [inline doc/user/search/index.md:105]: This is probably too late - but it seems weird to me to have a member of `experimentalFeatures` that is not a boolean. The options for an experimental feature usually don't go in here I think, only the enablement

File: doc/user/search/index.md [hunk 5]
```diff
@@ -92,6 +92,36 @@ Examples:
 
 Tip: On the statistics page, you can enter an empty query to see statistics across all repositories.
 
+
+### Version contexts
+
+Many companies have old versions of code running in production and need to search across all the code for a specific release.
```
Comment (felixfbecker) [inline doc/user/search/index.md:98]: Suggestion: don't limit to only companies
```suggestion
Many organizations have old versions of code running in production and need to search across all the code for a specific release.
```

File: web/src/nav/VersionContextDropdown.tsx [hunk 6]
```diff
@@ -111,10 +111,13 @@ export const VersionContextDropdown: React.FunctionComponent<VersionContextDropd
...
                                             <p className="mb-2">
-                                                Version contexts allow you to search a set of repositories based on a
-                                                hash, tag, or other interesting moment in time of multiple code bases.
-                                                Your administrator can configure version contexts in the site
-                                                configuration.
+                                                Version contexts (
+                                                <a href="http://docs.sourcegraph.com/user/search#version-contexts">
+                                                    documentation
+                                                </a>
+                                                ) allow you to search a set of repositories based on a hash, tag, or
```
Comment (attfarhan) [inline web/src/nav/VersionContextDropdown.tsx:118]:
```suggestion
                                                ) allow you to search a set of repositories based on a commit hash, tag, or
```

**Pull request merged**